dashboard/client/src/pages/serve/ServeEntityLogViewer.tsx

```diff
@@ -0,0 +1,287 @@
+import {
+  Box,
+  createStyles,
+  makeStyles,
+  MenuItem,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import _ from "lodash";
+import React, { useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
+import {
+  MultiTabLogViewer,
+  MultiTabLogViewerTabDetails,
+} from "../../common/MultiTabLogViewer";
+import { Section } from "../../common/Section";
+import { ServeDeployment, ServeSystemActor } from "../../type/serve";
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    captionText: {
+      color: theme.palette.grey[600],
+    },
+  }),
+);
+
+type ServeEntityLogsProps = {
+  controller?: ServeSystemActor;
+  proxies?: ServeSystemActor[];
+  deployments: ServeDeployment[];
+};
+
+/**
+ * A component that displays a log viewer for all types of Serve logs.
+ * A user uses dropdown menus to select the entity they wish to view logs of.
+ */
+export const ServeEntityLogViewer = ({
+  controller,
+  proxies,
+  deployments,
+}: ServeEntityLogsProps) => {
+  const classes = useStyles();
+
+  const [params, setParams] = useSearchParams();
+
+  const showEntityGroups = controller !== undefined || proxies !== undefined;
+  const defaultEntityGroupName = showEntityGroups
+    ? controller
+      ? "controller"
+      : "proxies"
+    : "deployments";
+
+  const selectedEntityGroupName =
+    params.get("entityGroup") || defaultEntityGroupName;
```

Comment:
the string vals for search params should be extracted to vars, it's hard to see who is setting this value